pyvista/core/utilities/geometric_objects.py

```diff
@@ -383,6 +395,532 @@ def Sphere(
+
+    radius = np.linspace(inner_radius, outer_radius, radius_resolution)
+    theta = np.linspace(start_theta, end_theta, theta_resolution)
+    phi = np.linspace(start_phi, end_phi, phi_resolution)
+    return SolidSphereGeneric(
+        radius, theta, phi, center, direction, radians=radians, tolerance=tolerance
+    )
+
+
+def SolidSphereGeneric(
+    radius=None,
+    theta=None,
+    phi=None,
+    center=(0.0, 0.0, 0.0),
+    direction=(0.0, 0.0, 1.0),
+    radians=False,
+    tolerance=1.0e-8,
+):
+    """Create a solid sphere with flexible sampling.
+
+    A solid sphere fills space in 3D in comparison to
+    :func:`pyvista.Sphere`, which is a 2D surface.
+
+    This function allows user defined sampling of each spherical
+    coordinate, whereas :func:`pyvista.SolidSphere`
+    only allows linear sampling.   Angles are by default
+    specified in degrees.
+
+    PyVista uses a convention where ``theta`` represents the azimuthal
+    angle (similar to degrees longitude on the globe) and ``phi``
+    represents the polar angle (similar to degrees latitude on the
+    globe). In contrast to latitude on the globe, here
+    ``phi`` is 0 degrees at the North Pole and 180 degrees at the South
+    Pole. ``phi=0`` is on the positive z-axis by default.
+    ``theta=0`` is on the positive x-axis by default.
+
+    Parameters
+    ----------
+    radius : sequence[float], optional
+        A monotonically increasing sequence of values specifying radial
+        points. Must have at least two points and be non-negative.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying ``theta``
+        points. Must have at least two points.  Can have any value as long
+        as range is within 360 degrees. Large magnitudes may result in
+        problems with endpoint overlap detection.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying ``phi``
+        points. Must have at least two points.  Must be between
+        0 and 180 degrees.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center coordinate vector in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction coordinate vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees ``phi``.
+
+    radians : bool, default: False
+        Whether to use radians for ``theta`` and ``phi``.
+
+    tolerance : bool, default: 1.0e-8
+        Absolute tolerance for endpoint detection for ``theta``, ``phi`,
+        and ``radius``. ``tolerance`` is always compared in radians for
+        angular quantities.
+
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.SolidSphere: Sphere creation using linear sampling.
+    pyvista.Sphere: Sphere that describes outer 2D surface.
+
+    Examples
+    --------
```

Comment:
@MatthewFlamm 
The example code should be the ideal minimum code for an introduction in my mind. I would appreciate it if you consider adding a script to the example directory to move the code around :)

Reply:
The first example here can be dropped.  I would prefer to keep both of the last two as the linear sampling case gives context for the nonlinear case.  I can streamline the text though.

Reply:
This is a complex feature so I don't mind more examples. Users can just not look at them if they don't want to. But it's highly nontrivial what this feature is about so it's valuable to users to see more examples I think.

Reply:
I think the comment was good, I dropped the example that just plotted with no user parameters.  This doesn't help distinguish what is different about this function compared to `SolidSphere`.  This leaves the essential examples as you point out.  I was trying to cook up an example with data like the density of the earth in different layers.  This would be better in a separate gallery example.